server/routes/creatingReports/whyWasUoFApplied.ts

```diff
@@ -5,52 +5,79 @@ import { paths } from '../../config/incident'
 
 const FORM = 'reasonsForUseOfForce'
 
+enum SubmitType {
+  SAVE_AND_CONTINUE = 'save-and-continue',
+  SAVE_AND_RETURN = 'save-and-return',
+}
+
 export default class WhyWasUoFAppliedRoutes {
   constructor(private readonly draftReportService: DraftReportService) {}
 
   public view() {
     return async (req: Request, res: Response): Promise<void> => {
       const { bookingId } = req.params
-      const { reasons } = await this.draftReportService.getSelectedReasonsForUoF(req.user.username, Number(bookingId))
+      const { isComplete, reasons } = await this.draftReportService.getUoFReasonState(
+        req.user.username,
+        Number(bookingId)
+      )
+
+      const selectedReasons = req.flash('clearingOutReasons')?.length ? [] : reasons
 
       return res.render('formPages/incident/select-uof-reasons', {
         errors: req.flash('errors'),
-        data: { UofReasons, reasons },
+        data: { bookingId, UofReasons, reasons: selectedReasons },
+        editMode: isComplete,
       })
     }
   }
 
   public submit(): RequestHandler {
     return async (req, res) => {
       const { bookingId } = req.params
-      const { reasons } = req.body
+      const { reasons, submitType } = req.body
 
       if (!reasons) {
+        req.flash('clearingOutReasons', true)
         req.flash('errors', [{ href: '#reasons', text: 'Select the reasons why use of force was applied' }])
         return res.redirect(req.originalUrl)
       }
 
-      await this.draftReportService.process(res.locals.user, Number(bookingId), FORM, { reasons })
+      if (submitType === SubmitType.SAVE_AND_RETURN) {
+        await this.draftReportService.process(res.locals.user, Number(bookingId), FORM, { reasons })
+        return res.redirect(paths.reportUseOfForce(bookingId))
+      }
```

Comment:
With multiple reasons, when the report isn't complete and the user clicks save and return we will allow them to do so by leaving the primary reason unanswered. 

Once the report is complete we do not allow them saving it in a partially complete form - any actual persistent update is only done once they have selected a primary reason